qa/tasks/pykmip.py

```diff
@@ -0,0 +1,520 @@
+                             'Accept': '*/*',
+                             'X-Auth-Token': token_id},
+                    body=key1_json
+                )
+            except:
+                log.info("catched exception!")
+                run_in_pykmip_dir(ctx, cclient, ['sleep', '900'])
+
+            pykmip_sec_resp = sec_req.getresponse()
+            if not (pykmip_sec_resp.status >= 200 and
+                    pykmip_sec_resp.status < 300):
+                raise Exception("Cannot create secret")
+            pykmip_data = json.loads(pykmip_sec_resp.read())
+            if 'secret_ref' not in pykmip_data:
+                raise ValueError("Malformed secret creation response")
+            secret_ref = pykmip_data["secret_ref"]
+            log.info("secret_ref=%s", secret_ref)
+            secret_url_parsed = urlparse(secret_ref)
+            acl_json = json.dumps(
+                {
+                    "read": {
+                        "users": [rgw_user_id],
+                        "project-access": True
+                    }
+                })
+            acl_req = httplib.HTTPConnection(secret_url_parsed.netloc, timeout=30)
+            acl_req.request(
+                'PUT',
+                secret_url_parsed.path+'/acl',
+                headers={'Content-Type': 'application/json',
+                         'Accept': '*/*',
+                         'X-Auth-Token': token_id},
+                body=acl_json
+            )
+            pykmip_acl_resp = acl_req.getresponse()
+            if not (pykmip_acl_resp.status >= 200 and
+                    pykmip_acl_resp.status < 300):
+                raise Exception("Cannot set ACL for secret")
+
+            key = {'id': secret_ref.split('secrets/')[1], 'payload': secret['base64']}
+            ctx.pykmip.keys[secret['name']] = key
+
+    run_in_pykmip_dir(ctx, cclient, ['sleep', '3'])
+    try:
+        yield
+    finally:
+        pass
+
+
+@contextlib.contextmanager
+def task(ctx, config):
+    """
+    Deploy and configure Keystone
+
+    Example of configuration:
+
+    tasks:
+      - local_cluster:
+          cluster_path: /home/adam/ceph-1/build
+      - local_rgw:
+      - tox: [ client.0 ]
+      - pykmip:
+          client.0:
+            force-branch: master
+            config:
+              clientca: ca-ssl-cert
+              servercert: pykmkp-ssl-cert-and-key
+            secrets:
+              - name: my-key-1
+                base64: a2V5MS5GcWVxKzhzTGNLaGtzQkg5NGVpb1FKcFpGb2c=
+              - name: my-key-2
+                base64: a2V5Mi5yNUNNMGFzMVdIUVZxcCt5NGVmVGlQQ1k4YWg=
+      - s3tests:
+          client.0:
+            force-branch: master
+            kms_key: my-key-1
+      - rgw:
+          client.0:
+            use-pykmip-role: client.0
+    """
+    assert config is None or isinstance(config, list) \
+        or isinstance(config, dict), \
+        "task keystone only supports a list or dictionary for configuration"
```

Comment:
task keystone?

Reply:
Yes, "task keystone".  I dunno, think "Japanese manga?"  It's exactly the same language used in 14 other teuthology tasks for about the same problem, so I think this should stay as is.

Reply:
ok